TradeEngine/src/main/java/com/gala/sam/tradeEngine/service/MarketService.java

```diff
@@ -48,63 +61,115 @@ void init() {
+  }
+
+  private void handleOrderWithTimer(AbstractOrder order, AbstractOrderProcessor orderProcessor)
+      throws OrderTimeInForceNotSupportedException, OrderDirectionNotSupportedException {
+    StopWatch orderProcessorTimer = new StopWatch();
+    orderProcessorTimer.start();
+    orderProcessor.process(order);
+    orderProcessorTimer.stop();
+    log.info("Order {} was handled in {} milliseconds", order.getOrderId(),
+        orderProcessorTimer.getTotalTimeMillis());
   }
 
-  private void processTriggeredStopOrders() {
-    Iterator<StopOrder> it = marketState.getStopOrders().iterator();
+  private void processTriggeredStopOrders() throws ProcessingStopOrderException {
+    Iterator<AbstractStopOrder> it = marketState.getStopOrders().iterator();
     while (it.hasNext()) {
-      StopOrder stopOrder = it.next();
-      log.info("Testing Trigger on: " + stopOrder.toString());
+      AbstractStopOrder stopOrder = it.next();
+      log.debug("Testing Trigger on: " + stopOrder.toString());
       if (isStopLossTriggered(stopOrder)) {
-        log.info("Stop OrderReq Triggered");
+        log.debug("Stop order request Triggered");
         it.remove();
         orderRepository.delete(stopOrder);
-        ActiveOrder activeOrder = stopOrder.toActiveOrder();
-        processOrder(activeOrder);
+        AbstractActiveOrder activeOrder = stopOrder.toActiveOrder();
+        try {
+          processOrder(activeOrder);
+        } catch (AbstractOrderFieldNotSupportedException e) {
+          throw new ProcessingStopOrderException(stopOrder, e);
+        }
       } else {
-        log.info("Stop OrderReq not Triggered");
+        log.debug("Stop order request not Triggered");
       }
     }
   }
 
-  private boolean isStopLossTriggered(StopOrder stopOrder) {
-    ActiveOrder activeOrder = stopOrder.toActiveOrder();
+  private boolean isStopLossTriggered(AbstractStopOrder stopOrder) {
+    AbstractActiveOrder activeOrder = stopOrder.toActiveOrder();
     Optional<Float> lastExec = marketState.getTickerQueueGroup(activeOrder)
         .getLastExecutedTradePrice();
     log.debug("Checking if there has been a previous trade");
     if (lastExec.isPresent()) {
       log.debug("Previous trade found, checking direction");
-      if (activeOrder.getDirection().equals(DIRECTION.BUY)) {
+      if (activeOrder.getDirection().equals(Direction.BUY)) {
         log.debug("Buy direction: testing trigger");
         return stopOrder.getTriggerPrice() <= lastExec.get();
-      } else if (activeOrder.getDirection().equals(DIRECTION.SELL)) {
+      } else if (activeOrder.getDirection().equals(Direction.SELL)) {
         log.debug("Sell direction: testing trigger");
         return stopOrder.getTriggerPrice() >= lastExec.get();
       } else {
-        throw new UnsupportedOperationException("OrderReq direction not supported");
+        throw new UnsupportedOperationException("order direction not supported");
```

Comment:
I really don't think that you need to throw an exception here. Just log a warning, then return false. (Logically, the stop order can't be triggered, because it can't be parsed.)